### PR TITLE
Initial Text Editor Implementation

### PR DESCRIPTION
**Pull request by rubywwwilde**
## Summary
  First major milestone: a functional block-based text editor built on CodeMirror with local-first architecture.

**Core editing:**
- Block-based document structure with recursive nesting
- Text editing via CodeMirror 6 integration
- Block splitting (Enter), merging (Backspace/Delete), indentation (Tab/Shift+Tab)
- Full cursor navigation (arrow keys) with pixel-based position preservation across blocks

**Architecture:**                                                                             - Effect-TS service layer with dependency injection (external → domain → UI tiers)
- LiveStore for node structure/ordering/UI state with event sourcing
- Yjs CRDT for text content (enables reliable undo, future collaboration)
- URL-based navigation with browser back/forward support

**Testing:**
- Comprehensive browser tests using BDD-style Given/When/Then helpers
- Tests for all keyboard operations and edge cases
                                                                                              ## NOT included (future iterations)
- Data export/import (needed before daily use)
- Sidebar with document tree
- Block-level selection
- Node type system (foundation for advanced features)
- Breadcrumbs navigation
- Starred documents
- Multiple workspaces

## Test plan
- [x] `pnpm -F @teloi/web test:browser` - all browser tests pass
- [x] `cd apps/web && pnpm tsc --noEmit` - zero TypeScript errors
- [x] Manual verification of core editing operations

## Summary by CodeRabbit

* **New Features**
  * Local-first block editor with browser persistence, CRDT sync, multi-pane layout, URL-driven workspace navigation, runtime startup, and a dynamic Table component.

* **Documentation**
  * New README, comprehensive developer guide, and a pre-commit review checklist.

* **Tests**
  * Large suite of end-to-end browser tests for navigation, selection, keyboard editing (arrows, Enter/Tab/Backspace/Delete/Undo/Zoom), whitespace, and table streaming.

* **Style**
  * Full theming and editor styling (light/dark tokens, spacing, typography, table styles).

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: apps/web/src/ui/Block.tsx [hunk 1]
```diff
@@ -0,0 +1,889 @@
...
+      isActive: false,
+      childBlockIds: [] as readonly Id.Block[],
+      selection: null as { anchor: number; head: number; goalX: number | null; goalLine: "first" | "last" | null; assoc: -1 | 1 | null } | null,
+    },
+  });
+
+  // Get Y.Text and UndoManager for this block's node
+  const [, nodeId] = Id.parseBlockId(blockId).pipe(
+    Effect.runSync,
+  );
```
[the remark this answers is not in the document]
Comment (rubywwwilde) [reply apps/web/src/ui/Block.tsx]: okay, it fails, but can we recover anyways?

File: apps/web/src/services/external/Store/index.ts [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        ...args: QueryParams<TResult>
+      ) => Effect.Effect<TResult> = (query) => {
+        try {
+          var result = Store.query(query);
+        } catch (e) {
+          console.error("Store query unsuccessful", query, e);
+          throw new Error("Store query unsuccessful", { cause: e });
+        }
+        return Effect.succeed(result);
+      };
```
[the remark this answers is not in the document]
Comment (rubywwwilde) [reply apps/web/src/services/external/Store/index.ts]: I don't have any expected errors or alternatives here. If it fails, it means there is bug in the code.
